src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -270,6 +272,8 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
                     }
```

Comment:
The loop is looking up by symbol. Not sure if looking up by location would be good enough. Will test.